### DISCUSSION TITLE
Recording streams on my NAS with Ubuntu Server 22.04.3 LTS

### DISCUSSION THREAD

**Discussion by KyodaiKen** [streamlink/streamlink#5621] [Q&A] [answered]
### Description

Hi all,

I would like to know if it is possible to record streams using streamlink without a desktop environment.

When I try to run `apt install streamlink` it wants to install a desktop enviornment.

But I would like to schedule recordings on my NAS without a desktop environment, as it isn't needed. Recordings will be played on any client of my NAS.

Is there a way to install a minimalistic streamlink version to achieve this?

Thank you all for helping me!

**Comment by bastimeyer** [comment, marked as answer]
Streamlink's debian package adds `vlc`, `mpv` or `mplayer` as **recommended** packages, which in turn pull the entire desktop stack:
https://packages.debian.org/sid/streamlink

`apt` and `apt-get` by default always install recommended packages unless you set `--no-install-recommends`. See the manual:
https://manpages.debian.org/stretch/apt/apt-get.8.en.html

Depending on your distro however, installing the Streamlink Debian package is not recommended, as you'll most likely only find outdated package versions:
https://streamlink.github.io/install.html#package-availability

**Reply by mkbloke** [reply]
I was just starting to answer this myself.  Consider using AppImage or Python pip (preferably in a virtual environment) as options to install Streamlink.  See the info on the install page linked above for all the Linux install options.

**Reply by KyodaiKen** [reply]
Yeah... Ubuntu/Debian in a nutshell. Packages from 3000 BC. But I got the motto: Never change a running system. I will try pip instead. Why is a virtual environment recommended? The NAS cannot be reached from the internet. I got no ports open.

**Reply by bastimeyer** [reply]
virtual environment = separate python package environment which doesn't mix with your system's global python package environment

Check the linked docs.

**Reply by mkbloke** [reply]
> Why is a virtual environment recommended?

It isolates the required Streamlink dependencies to the Streamlink installation environment.  Other virtual envs will isolate their dependencies in their own environment, separate from other virtual envs.

**Reply by KyodaiKen** [reply]
Alright. It's approaching midnight here. I will do that tomorrow! Thanks everyone!

**Reply by KyodaiKen** [reply]
The install instructions using `virtualenv` and `pip` worked like a charme! Thank you!
